shared/src/main/scala/tastyquery/reader/TreeUnpickler.scala

```diff
@@ -428,200 +455,219 @@ class TreeUnpickler(protected val reader: TastyReader, nameAtRef: NameTable) {
+        val args = readTerms(end)
+        // Construct Span from fn and args manually for infix expressions like `x < 0`
+        if (curSpan == NoSpan && fn.span != NoSpan) {
+          val argsSpan = args.foldLeft(NoSpan)((x, t) => x.union(t.span))
+          curSpan = fn.span.union(argsSpan)
+        }
+        Apply(fn, args)(curSpan)
+      case NAMEDARG =>
         reader.readByte()
-        new InlineMatch(readTerm, readCases[CaseDef](CaseDefFactory, end))
-      } else Match(readTerm, readCases[CaseDef](CaseDefFactory, end))
-    case BIND =>
-      val start = reader.currentAddr
-      reader.readByte()
-      val end = reader.readEnd()
-      val name = readName
-      // TODO: use type
-      val typ = readType
-      val term = readTerm
-      skipModifiers(end)
-      val symbol = ctx.getSymbol(start, RegularSymbolFactory)
-      Bind(name, term, symbol).definesTreeOf(symbol)
-    case ALTERNATIVE =>
-      reader.readByte()
-      val end = reader.readEnd()
-      Alternative(reader.until(end)(readTerm))
-    case UNAPPLY =>
-      reader.readByte()
-      val end = reader.readEnd()
-      val fun = readTerm
-      val args = reader.collectWhile(reader.nextByte == IMPLICITarg)({
-        assert(reader.readByte() == IMPLICITarg, posErrorMsg)
-        readTerm
-      })
-      // TODO: use pattern type
-      val patType = readType
-      val patterns = reader.until(end)(readTerm)
-      Unapply(fun, args, patterns)
-    case REPEATED =>
-      reader.readByte()
-      val end = reader.readEnd()
-      val elemType = readTypeTree
-      SeqLiteral(reader.until(end)(readTerm), elemType)
-    case WHILE =>
-      reader.readByte()
-      reader.readEnd()
-      While(readTerm, readTerm)
-    case RETURN =>
-      reader.readByte()
-      val end = reader.readEnd()
-      val from = readSymRef
-      val expr = reader.ifBefore(end)(readTerm, EmptyTree)
-      // TODO: always just taking the name?
-      // return always returns from a method, i.e. something with a TermName
-      Return(expr, TermRefTree(from.name.asInstanceOf[TermName], TermRef(NoPrefix, from)))
-    case INLINED =>
-      reader.readByte()
-      val end = reader.readEnd()
-      val expr = readTerm
-      val caller: TypeIdent =
-        reader.ifBefore(end)(
-          tagFollowShared match {
-            // The caller is not specified, this is a binding (or next val or def)
-            case VALDEF | DEFDEF => EmptyTypeIdent
-            case _               => readTypeTree.asInstanceOf[TypeIdent]
-          },
-          EmptyTypeIdent
-        )
-      val bindings = reader.until(end)(readValOrDefDef)
-      Inlined(expr, caller, bindings)
-    case SHAREDterm =>
-      reader.readByte()
-      forkAt(reader.readAddr()).readTerm
+        NamedArg(readName, readTerm)(curSpan)
+      case TYPEAPPLY =>
+        reader.readByte()
+        val end = reader.readEnd()
+        val fnSpan = span
+        val fn = readTerm
+        val paramsSpan = span
```

Comment:
this would only get the span for the first type parameter, (all type trees have spans as well btw)

Reply:
Good to know. I'll add spans to `TypeTree`.

Reply:
Fixed